sound/hda/hdac_stream.c

```diff
@@ -405,14 +406,20 @@ int snd_hdac_stream_setup_periods(struct hdac_stream *azx_dev)
 {
 	struct hdac_bus *bus = azx_dev->bus;
 	struct snd_pcm_substream *substream = azx_dev->substream;
+	struct snd_compr_stream *cstream = azx_dev->cstream;
 	struct snd_pcm_runtime *runtime;
 	struct snd_dma_buffer *dmab;
 	__le32 *bdl;
 	int i, ofs, periods, period_bytes;
 	int pos_adj, pos_align;
 
-	runtime = substream->runtime;
-	dmab = snd_pcm_get_dma_buf(substream);
+	if (substream) {
+		runtime = substream->runtime;
+		dmab = snd_pcm_get_dma_buf(substream);
+	} else {
+		runtime = NULL;
```

Comment:
why not just initialize runtme to NULL instead?

Reply:
This and several other changes were requested by @plbossart i.e. move to more explicit assignments, and thus you got 'else' there. If you go back to probes PR, you would see that in previous patches indeed locals were initialized directly during declaration.
Matter of taste, I'd say. If you want to pursue the change, please agree on one solution with Pierre.